rmf_site_format/src/model.rs

```diff
@@ -57,3 +58,85 @@ impl Default for Model {
         }
     }
 }
+
+/// Defines a property in a model description, that will be added to all instances
+#[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "bevy", derive(Component, Reflect))]
+pub struct ModelProperty<T: Default + Clone>(pub T);
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum OptionalModelProperty {
+    DifferentialDrive(DifferentialDrive),
+    MobileRobotMarker(MobileRobotMarker),
+    Tasks(Tasks<Entity>),
```

Comment:
I think this is making CI fail, the moment we compile `rmf_site_format` without the `bevy` feature `Entity` will not be defined, you can try this locally by just running `cargo build` in the `rmf_site_format` folder.
I don't know the details of the `Tasks` implementation but usually if we need to reference an entity in a "non bevy specific way" in `rmf_site_format` we just use `RefTrait`:
https://github.com/open-rmf/rmf_site/blob/7d21b4881ef4aeb251e93733388117c2a3275ab2/rmf_site_format/src/site.rs#L224

Which is a trait for references and is implemented for `Entity` (for usage in site editor) and `u32` (for usage in serialization / deserialization, as well as not having a bevy dependency).

Now the problem with this approach is that it will make `OptionalModelProperty` generic and that will propagate all the way up to the `ModelDescriptionBundle`, which might not be what you want

Reply:
But again, maybe having to do `ModelDescriptionBundle<u32>` instead of `ModelDescriptionBundle` wouldn't be too bad

Reply:
Ah thank you for pointing this out! I couldn't figure out the underlying issue behind using `Entity`. I can see another (and maybe better) way to store location information in tasks without using `Entity`, especially since the current method to save tasks involve exporting the entity ID that could change between runs. Let me make that change.

Reply:
Yap that's the classic use for `RefTrait`, to use both `Entity` and `u32`, usually the `u32` that is saved / loaded is the value that is contained in the `SiteID` component

Reply:
Fixed in [347d546](https://github.com/open-rmf/rmf_site/pull/242/commits/347d546e4b8e522622464d93a069665a7f76d5a6)